src/index.js

```diff
@@ -3,103 +3,105 @@ const http = require('http')
 const express = require('express')
 const socketio = require('socket.io')
 const Filter = require('bad-words')
-const { generateMessage, generateLocationMessage } = require('./utils/messages')
+
+const {
+    generateWelcome,
+    generateJoined,
+    generateLeft,
+    makeMessageGenerator,
+} = require('./utils/messages')
+
 const {
-    addUser,
-    removeUser,
-    getUser,
-    getUsersInRoom,
-    getRooms,
-} = require('./utils/users')
+    userToRoomJoiner,
+    roomDataFetcher,
+    userFromRoomDropper,
+} = require('./services')
+const { roomRepo } = require('./repos')
+const { Location } = require('./entities')
 
 const app = express()
 const server = http.createServer(app)
 const io = socketio(server)
 
-const port = process.env.PORT
-
 const publicDirPath = path.join(__dirname, '../public')
+const filter = new Filter()
 
 app.use(express.static(publicDirPath))
 
-app.get('/rooms', async (req, res) => {
+app.get('/rooms', (req, res) => {
     try {
-        const rooms = getRooms()
-        res.send({ rooms })
+        const rooms = roomRepo.all
+        res.send({ rooms: rooms.map((r) => r.title) })
     } catch (error) {
+        console.error(error)
         res.status(500).send({ error: error.message })
     }
 })
 
 io.on('connection', (socket) => {
     console.log('New WebSocket connection.')
+    let context = {
+        user: null,
+        room: null,
+    }
 
-    socket.on('join', ({ username, room }, callback) => {
-        const { error, user } = addUser({ id: socket.id, username, room })
+    let generateMessage
+    let generateLocationMessage
 
-        if (error) {
-            return callback(error)
-        }
+    socket.on('join', async ({ username, room: roomTitle }, callback) => {
+        const { error, user, room } = await userToRoomJoiner.call({
+            username,
+            roomTitle,
+        })
+        if (error) return callback(error)
 
-        socket.join(user.room)
+        context = {
+            user,
+            room,
+        }
+        generateMessage = makeMessageGenerator(user)
+        generateLocationMessage = makeMessageGenerator(
+            user,
+            (location) => `${location}`,
+            'url'
+        )
 
-        socket.emit('message', generateMessage('Admin', 'Welcome!'))
-        socket.broadcast
-            .to(user.room)
-            .emit(
-                'message',
-                generateMessage('Admin', `${user.username} has joined!`)
-            )
+        socket.join(room)
+        socket.emit('message', generateWelcome())
+        socket.broadcast.to(room).emit('message', generateJoined(user))
 
-        io.to(user.room).emit('roomData', {
-            room: user.room,
-            users: getUsersInRoom(user.room),
-        })
+        const roomWithUsers = await roomDataFetcher.call(room)
```

Comment:
Why does `roomDataFetcher` start with a lower case letter?
I mean yes, you require it by files name and there it does start with a lower case letter....but the class name starts with an upper case letter?
How does this particular require work?

Reply:
see `services/index.js`